common/src/db/query.rs

```diff
@@ -130,42 +145,210 @@ pub enum Error {
 // Internal types
 /////////////////////////////////////////////////////////////////////////
 
+#[derive(Default)]
+pub struct Columns {
+    columns: Vec<(ColumnRef, ColumnDef)>,
+}
+
+impl Columns {
+    pub fn with_entity_columns<E: EntityTrait>() -> Self {
+        let mut columns = Columns::default();
+        columns.add_columns::<E::Column>();
+        columns
+    }
+
+    pub fn add_columns<C: ColumnTrait>(&mut self) -> &mut Self {
+        for c in C::iter() {
+            let (t, u) = c.clone().as_column_ref();
+            let column_ref = ColumnRef::TableColumn(t, u);
+            let column_def = c.def();
+            self.columns.push((column_ref, column_def));
+        }
+        self
+    }
+
+    pub fn add_column<I: IntoIdentity>(&mut self, name: I, def: ColumnDef) -> &mut Self {
+        self.columns
+            .push((name.into_identity().into_column_ref(), def));
+        self
+    }
+
+    pub fn for_field(&self, field: &str) -> Option<(ColumnRef, ColumnDef)> {
+        self.columns
+            .iter()
+            .find(|(col_ref, col_def)| {
+                matches!( col_ref,
+                   ColumnRef::Column(name)
+                    | ColumnRef::TableColumn(_, name)
+                    | ColumnRef::SchemaTableColumn(_, _, name)
+                        if name.to_string() == field)
+            })
+            .cloned()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &(ColumnRef, ColumnDef)> {
+        self.columns.iter()
+    }
+}
+
 #[derive(Debug)]
-struct Filter<T: EntityTrait> {
-    operands: Operand<T>,
+struct Filter {
+    operands: Operand,
     operator: Operator,
 }
 
-struct Sort<T: EntityTrait> {
-    field: T::Column,
+impl Filter {
```

Comment:
It's hard for me to see what changed here. I think it'd be easier to see if you edited the old code directly, instead of commenting it. (I kinda hate seeing huge commented-out sections of code anywhere)